src/snowflake/snowpark/dataframe.py

```diff
@@ -1175,14 +1176,16 @@ def select(
             raise ValueError("The input of select() cannot be empty")
 
         # AST.
-        if _ast_stmt is None:
-            stmt = self._session._ast_batch.assign()
-            ast = with_src_position(stmt.expr.sp_dataframe_select__columns, stmt)
-            self.set_ast_ref(ast.df)
-            ast.variadic = is_variadic
-        else:
-            stmt = _ast_stmt
-            ast = None
+        stmt = None
+        if not _suppress_ast:
+            if _ast_stmt is None:
```

Comment:
Minor nit: I found it easier to add an `_emit_ast` parameter instead and change the first condition to `if _ast_stmt is None and _emit_ast`. This avoids the double negative and the additional changes to the logic in `select`, `filter`, and `distinct` later.

Either approach accomplishes the same thing though, so it's up to our preference for how we want to suppress AST generation internally, since this will likely be needed across many additional APIs.

Reply:
I have no strong preference over `_supress_ast` or `emit_ast`. However, we should keep one. I used `_supress_ast` to keep the naming similar to existing APIs (e.g., table).